lib/elixir/lib/system.ex

```diff
@@ -73,11 +104,30 @@ defmodule System do
   @doc """
   Elixir build information.
 
-  Returns a keyword list with Elixir version, git short revision hash and compilation date.
+  Returns a keyword list with Elixir version, compilation date, git tag and short revision hash.
   """
   @spec build_info() :: map
   def build_info do
-    %{version: version, date: get_date, revision: get_revision}
+    %{  date:     get_date,
+        revision: binary_part(get_revision, 0, 7),
+        tag:      get_tag,
+        version:  get_version,
+      }
+  end
+
+  @doc false
+  # Returns a string of the build info
+  @spec build_info(atom :: :version_build) :: String.t
+  def build_info(opt) when opt in [:version_build] do
+    case opt do
+      :version_build ->
+        info = build_info()
+        {:ok, v} = Version.parse(info[:version])
+        case v.pre do
```

Comment:
This would also show the revision for non -dev stuff, like -rc.0, which we don't want. In general, I am :-1: on this feature. Folks can easily access it if they want by calling System.build_info.